Mathlib/CategoryTheory/Monoidal/DayConvolution.lean

```diff
@@ -0,0 +1,208 @@
+    h.convolution ≅ h'.convolution :=
+  Functor.leftKanExtensionUnique h.convolution h.unit h'.convolution h'.unit
+
+@[simp]
+lemma uniqueUpToIso_hom_unit (h : DayConvolution F G) (h' : DayConvolution F G) :
+    h.unit ≫ CategoryTheory.whiskerLeft (tensor C) (h.uniqueUpToIso h').hom = h'.unit := by
+  simp [uniqueUpToIso]
+
+@[simp]
+lemma uniqueUpToIso_inv_unit (h : DayConvolution F G) (h' : DayConvolution F G) :
+    h'.unit ≫ CategoryTheory.whiskerLeft (tensor C) (h.uniqueUpToIso h').inv = h.unit := by
+  simp [uniqueUpToIso]
+
+variable (F G) [DayConvolution F G]
+
+section unit
+
+variable {x x' y y' : C}
+
+@[reassoc (attr := simp)]
+lemma unit_naturality (f : x ⟶ x') (g : y ⟶ y') :
+    (F.map f ⊗ₘ G.map g) ≫ (unit F G).app (x', y') =
+    (unit F G).app (x, y) ≫ (F ⊛ G).map (f ⊗ₘ g) := by
+  simpa [tensorHom_def] using (unit F G).naturality ((f, g) : (x, y) ⟶ (x', y'))
+
+variable (y) in
+@[reassoc (attr := simp)]
+lemma unit_naturality_id_right (f : x ⟶ x') :
+    F.map f ▷ G.obj y ≫ (unit F G).app (x', y) =
+    (unit F G).app (x, y) ≫ (F ⊛ G).map (f ▷ y) := by
+  simpa [tensorHom_def] using (unit F G).naturality ((f, 𝟙 _) : (x, y) ⟶ (x', y))
+
+variable (x) in
+@[reassoc (attr := simp)]
+lemma unit_naturality_id_left (g : y ⟶ y') :
+    F.obj x ◁ G.map g ≫ (unit F G).app (x, y') =
+    (unit F G).app (x, y) ≫ (F ⊛ G).map (x ◁ g) := by
+  simpa [tensorHom_def] using (unit F G).naturality ((𝟙 _, g) : (x, y) ⟶ (x, y'))
+
+end unit
+
+variable {F G}
+
+section map
+
+variable {F' G' : C ⥤ V} [DayConvolution F' G']
+
+/-- The morphism between day convolutions (provided they exist) induced by a pair of morphisms. -/
+def map (f : F ⟶ F') (g : G ⟶ G') : F ⊛ G ⟶ F' ⊛ G' :=
+  Functor.descOfIsLeftKanExtension (F ⊛ G) (unit F G) (F' ⊛ G') <|
+    (externalProductBifunctor C C V).map ((f, g) : (F, G) ⟶ (F', G')) ≫ unit F' G'
+
+variable (f : F ⟶ F') (g : G ⟶ G') (x y : C)
+
+@[reassoc (attr := simp)]
+lemma map_unit_app :
+  (unit F G).app (x, y) ≫ (map f g).app (x ⊗ y : C) =
+    (f.app x ⊗ₘ g.app y) ≫ (unit F' G').app (x, y) := by
+  simpa [tensorHom_def] using
+    (Functor.descOfIsLeftKanExtension_fac_app (F ⊛ G) (unit F G) (F' ⊛ G') <|
+      (externalProductBifunctor C C V).map ((f, g) : (F, G) ⟶ (F', G')) ≫ unit F' G') (x, y)
+
+end map
+
+variable (F G)
+/-- The universal property of left Kan extensions characterizes the functor
+corepresented by `F ⊛ G`. -/
+@[simps!]
+def corepresentableIso : coyoneda.obj (.op <| F ⊛ G) ≅
+    (whiskeringLeft _ _ _).obj (tensor C) ⋙ coyoneda.obj (.op <| F ⊠ G) :=
+  NatIso.ofComponents
+    (fun H ↦ Equiv.toIso <| Functor.homEquivOfIsLeftKanExtension _ (unit F G) _)
+
+/-- The universal property of left Kan extensions characterizes the functor
+corepresented by `F ⊛ G`. -/
+def corepresentable :
+    (whiskeringLeft _ _ _).obj (tensor C) ⋙ coyoneda.obj (.op <| F ⊠ G)|>.CorepresentableBy
+      (F ⊛ G) :=
+  Functor.corepresentableByEquiv.symm <| corepresentableIso F G
```

Comment:
Maybe it would be slightly better to define directly `corepresentableBy` (because the constructor takes a family of equivs, which we have). Then, it is unclear we should keep `corepresentableIso`?